### ISSUE TITLE
Rewrite overview

### ISSUE THREAD

**Issue by martinthomson** [quicwg/base-drafts#410]
A protocol overview should concentrate on establishing the function of the protocol at a high level.  That is, it should describe what the protocol is for and then how the protocol sets out to achieve those goals, with pointers to more detailed explanations of each.  It should act as a way for new readers to orient themselves in both the protocol and its specification.

The overview is currently pure self-promotion, which covers part of the first goal, but doesn't do much to help the reader understand how each of these wonderful *features* are actually put together.

**Comment by seanturner** [comment]
I tend to agree the front matter is some good marketing, but maybe we could retain it somehow by moving some of it to an informative annex?
